crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs

```diff
@@ -817,7 +1108,12 @@ impl<F> TryFrom<&DeutschebankRouterData<&RefundsRouterData<F>>> for Deutschebank
     fn try_from(item: &DeutschebankRouterData<&RefundsRouterData<F>>) -> Result<Self, Self::Error> {
         Ok(Self {
             changed_amount: item.amount.to_owned(),
-            kind: DeutschebankTransactionKind::Directdebit,
+            kind: match item.router_data.is_three_ds()
+                && matches!(item.router_data.payment_method, PaymentMethod::Card)
+            {
+                true => DeutschebankTransactionKind::Creditcard3ds20,
+                false => DeutschebankTransactionKind::Directdebit,
+            },
```

Comment:
Add explicit matching where
if sepa then value should be Directdebit
if cards + 3ds the value should be Creditcard3ds20
else not supported error